### PR TITLE
powerplatform_environment resource should support the cadence, administration mode, description and environment group parameters

### PR DESCRIPTION
**Pull request by mawasile**
This pull request includes several updates to the Power Platform provider, focusing on adding new attributes, modifying test cases, and updating workflow configurations. The most important changes include adding new fields to resources and data sources, updating test cases to reflect these changes, and modifying the GitHub Actions workflow for better coverage.

### Enhancements to Resources and Data Sources:
* Added `description` and `cadence` attributes to `powerplatform_environment` resource in `docs/resources/environment.md` and `examples/resources/powerplatform_environment/resource.tf`. [[1]](diffhunk://#diff-8136f433eee35bd9699d3341e876b4fc48589d95deaaf1a713a8ea2f4e1b7fcdR40-R44) [[2]](diffhunk://#diff-8136f433eee35bd9699d3341e876b4fc48589d95deaaf1a713a8ea2f4e1b7fcdR67-R69) [[3]](diffhunk://#diff-b41ec35c3b7292b3415740a208594ad759b440e07a3852c51a991353e6bfd183R15-R27)
* Added `administration_mode_enabled` and `background_operation_enabled` attributes to `dataverse` in `docs/resources/environment.md` and `docs/data-sources/environments.md`. [[1]](diffhunk://#diff-694fc2dad3ddffad89ec5f598a266e133f867c8ed152d10bde221047113899b4R79-R80) [[2]](diffhunk://#diff-8136f433eee35bd9699d3341e876b4fc48589d95deaaf1a713a8ea2f4e1b7fcdR87-R88)
* Updated `internal/powerplatform/services/environment/models_environment.go` to include new fields in the DTOs and conversion functions. [[1]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cL37-R47) [[2]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cR79) [[3]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cR98-R102) [[4]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cR123-R124) [[5]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cR214-R215) [[6]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cR233-R234) [[7]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cR245) [[8]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cR287-R296) [[9]](diffhunk://#diff-f8c1f0f2d7ab879839222d140074edd9b231a90f80b4cdd28ebe51a46fc4021cR361-R367)

### Test Case Updates:
* Updated acceptance and unit tests in `internal/powerplatform/services/environment/datasource_environments_test.go` to include checks for `description` and `cadence` attributes. [[1]](diffhunk://#diff-3a434702c91536f9718ea7711e5a190882f92ff98b2356eac136dbc56e6fbacaR26-R30) [[2]](diffhunk://#diff-3a434702c91536f9718ea7711e5a190882f92ff98b2356eac136dbc56e6fbacaR47-R48) [[3]](diffhunk://#diff-3a434702c91536f9718ea7711e5a190882f92ff98b2356eac136dbc56e6fbacaR101-R102) [[4]](diffhunk://#diff-3a434702c91536f9718ea7711e5a190882f92ff98b2356eac136dbc56e6fbacaR122-R123)

### Workflow Configuration:
* Modified `.github/workflows/run_tests.yml` to run tests on all Go files and updated the test command to run only acceptance tests. [[1]](diffhunk://#diff-c2e9f7c1bfaf65da79bbd29ed3ce389b2acc8b4099a257914a7292a66e6b9bc9R11) [[2]](diffhunk://#diff-c2e9f7c1bfaf65da79bbd29ed3ce389b2acc8b4099a257914a7292a66e6b9bc9L141-R142)

### Code Improvements:
* Removed unnecessary check for unknown plan values in `internal/powerplatform/modifiers/require_replace_object_to_empty_modifier.go`.
* Added new imports for validators in `internal/powerplatform/services/environment/datasource_environments.go`. [[1]](diffhunk://#diff-8f97cc00838eea12861c59580d707cf9662f94687720b4c5752abfb43780f073R11-R14) [[2]](diffhunk://#diff-8f97cc00838eea12861c59580d707cf9662f94687720b4c5752abfb43780f073R94-R104) [[3]](diffhunk://#diff-8f97cc00838eea12861c59580d707cf9662f94687720b4c5752abfb43780f073R115-R122)

These changes enhance the functionality and maintainability of the Power Platform provider by adding new features and improving test coverage.

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -128,7 +128,7 @@ jobs:
     - name: Run tests
       run: |
         export TF_CLI_CONFIG_FILE="${GITHUB_WORKSPACE}/deps/terraform.rc"
-        export TF_LOG=WARN
+        export TF_LOG=TRACE
```
Comment (mattdot) [inline .github/workflows/run_tests.yml:131]: Intentional to change this?

File: .github/workflows/run_tests.yml [hunk 2]
```diff
@@ -138,7 +138,7 @@ jobs:
         export ARM_TENANT_ID='${{ secrets.ACCEPTANCE_TESTS_ENV_TENANT_ID }}'
         export ARM_SUBSCRIPTION_ID='${{ secrets.ACCEPTANCE_TESTS_ENV_SUBSCRIPTION_ID }}'
         go clean -testcache
-        go test -p 10 -failfast -v ./... -run ^Test -coverprofile=test-coverage.out -timeout 300m
+        go test -p 10 -failfast -v ./... -run ^TestAccEnvironment -coverprofile=test-coverage.out -timeout 300m
```
Comment (mattdot) [inline .github/workflows/run_tests.yml:141]: Don't we want to run all the tests?

Comment (mawasile) [reply .github/workflows/run_tests.yml:141]: This is until the branch is ready for review

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (mawasile) [reply .github/workflows/run_tests.yml:131]: Yes

**Review by eduardodfmex — DISMISSED** [review]

**Review by eduardodfmex — DISMISSED** [review]

File: internal/powerplatform/services/environment/resource_environment.go [hunk 3]
```diff
@@ -117,6 +128,21 @@ func (r *EnvironmentResource) Schema(ctx context.Context, req resource.SchemaReq
...
 				Attributes: map[string]schema.Attribute{
+					"environment_group_id": schema.StringAttribute{
+						MarkdownDescription: "Unique environment group id (guid) that the environment belongs to. Empty guid `00000000-0000-0000-0000-000000000000` is considered as no environment group.",
+						Computed:            true,
+						Optional:            true,
+					},
+					"administration_mode_enabled": schema.BoolAttribute{
+						MarkdownDescription: "Select to enable administration mode for the environment. See [Admin mode](https://learn.microsoft.com/en-us/power-platform/admin/admin-mode) for more information. ",
+						Optional:            true,
+						Computed:            true,
```
Comment (mattdot) [inline internal/powerplatform/services/environment/resource_environment.go:139]: Should be optional with a default and not computed.

File: internal/powerplatform/services/environment/resource_environment.go [hunk 4]
```diff
@@ -67,6 +68,16 @@ func (r *EnvironmentResource) Schema(ctx context.Context, req resource.SchemaReq
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: "Description of the environment",
+				Optional:            true,
+				Computed:            true,
```
Comment (mattdot) [inline internal/powerplatform/services/environment/resource_environment.go:74]: Should be optional with a default

File: internal/powerplatform/services/environment/resource_environment.go [hunk 5]
```diff
@@ -67,6 +68,16 @@ func (r *EnvironmentResource) Schema(ctx context.Context, req resource.SchemaReq
...
+			},
+			"cadence": schema.StringAttribute{
+				MarkdownDescription: "Cadence of updates for the environment (Frequent, Moderate)",
+				Optional:            true,
+				Computed:            true,
```
Comment (mattdot) [inline internal/powerplatform/services/environment/resource_environment.go:79]: Should be optional with a default

File: internal/powerplatform/services/environment/resource_environment.go [hunk 6]
```diff
@@ -67,6 +68,16 @@ func (r *EnvironmentResource) Schema(ctx context.Context, req resource.SchemaReq
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: "Description of the environment",
+				Optional:            true,
+				Computed:            true,
+			},
+			"cadence": schema.StringAttribute{
+				MarkdownDescription: "Cadence of updates for the environment (Frequent, Moderate)",
```
Comment (mattdot) [inline internal/powerplatform/services/environment/resource_environment.go:77]: Documentation link?

File: internal/powerplatform/services/environment/resource_environment.go [hunk 7]
```diff
@@ -117,6 +128,21 @@ func (r *EnvironmentResource) Schema(ctx context.Context, req resource.SchemaReq
 					modifiers.RequireReplaceObjectToEmptyModifier(),
 				},
 				Attributes: map[string]schema.Attribute{
+					"environment_group_id": schema.StringAttribute{
+						MarkdownDescription: "Unique environment group id (guid) that the environment belongs to. Empty guid `00000000-0000-0000-0000-000000000000` is considered as no environment group.",
+						Computed:            true,
+						Optional:            true,
+					},
```
Comment (mattdot) [inline internal/powerplatform/services/environment/resource_environment.go:135]: Consider moving this to top level property.

File: internal/powerplatform/services/environment/resource_environment.go [hunk 8]
```diff
@@ -386,6 +425,16 @@ func (r *EnvironmentResource) Update(ctx context.Context, req resource.UpdateReq
...
+		environmentDto.Properties.Description = plan.Description.ValueString()
+	}
+
+	if !plan.Cadence.IsNull() && plan.Cadence.ValueString() != "" {
+		environmentDto.Properties.UpdateCadence = &UpdateCadenceDto{
+			Id: plan.Cadence.ValueString(),
+		}
+	}
+
 	if !plan.BillingPolicyId.IsNull() && plan.BillingPolicyId.ValueString() != "" {
```
[the remark this answers is not in the document]
Comment (mawasile) [reply internal/powerplatform/services/environment/resource_environment.go:438]: works ok

**Review by mattdot — APPROVED** [review]

**Pull request merged**